monkey/common/agent_events/abstract_agent_event.py

```diff
@@ -24,7 +24,7 @@ class AbstractAgentEvent(InfectionMonkeyBaseModel, ABC):
         :param tags: The set of tags associated with the event
     """
 
-    source: AgentID
+    source: AgentID = Field(default=None)
```

Comment:
Since the field is not `Optional`, I don't think it makes sense to default to `None`